src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -59,6 +59,13 @@ lemma component_compl_mk_eq_of_adj (G : simple_graph V) {v w : V} (vK : v ∉ K)
   (a : G.adj v w) : G.component_compl_mk vK = G.component_compl_mk wK :=
 by { rw [connected_component.eq], apply adj.reachable, exact a }
 
+/--
+In an infinite graph, the set of components out of a finite set is nonempty.
+-/
+lemma component_compl_nonempty_of_infinite (G : simple_graph V) [infinite V] (K : finset V) :
+  nonempty (G.component_compl K) :=
```

Comment:
If this is `_root_.nonempty`, is there a reason this isn't an instance?

Reply:
I guess the same comment that Yael made for `finite` applies: I'll make it into an instance, thanks!